src/cdk/explorer/nodes/propertyNode.ts

```diff
@@ -0,0 +1,52 @@
+/*!
+ * Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { ext } from '../../../shared/extensionGlobals'
+import { AWSTreeNodeBase } from '../../../shared/treeview/nodes/awsTreeNodeBase'
+
+/*
+ * Represents a property of a CDK construct. Properties can be simple key-value pairs, Arrays,
+ * or objects that are nested deeply
+ */
+export class PropertyNode extends AWSTreeNodeBase {
+    public constructor(
+        public readonly label: string,
+        public readonly collapsibleState: vscode.TreeItemCollapsibleState,
+        public readonly children?: { [key: string]: any }
+    ) {
+        super(label, collapsibleState)
+        this.contextValue = 'awsCdkPropertyNode'
+        this.iconPath = {
+            dark: vscode.Uri.file(ext.iconPaths.dark.settings),
+            light: vscode.Uri.file(ext.iconPaths.light.settings)
+        }
+    }
+
+    public async getChildren(): Promise<PropertyNode[]> {
+        if (!this.children) {
+            return []
+        }
+
+        return generatePropertyNodes(this.children)
```

Comment:
due to the nested nature of properties, that's the intention (to redraw). As CDK authors configure some of those properties, they'll change in depth.

I was also thinking ahead of what we'd want to update when we add watchers on the tree files eventually.

Let me know if I'm thinking about this wrong and there's a more elegant way to go about it.

Reply:
What drives refreshes of the explorer? Closing/reopening nodes, a refresh button, some sort of automated refresh logic, etc.? I think this implies closing/reopening a node, which I'm not sure is the expected experience. I'd say have it fall in line with what you're setting the users up to expect going forward.

Reply:
when a node is expanded, it's `getChildren` will only be called once. After that closing/reopening it will not trigger it. A refresh however will also call `getChildren` on all nodes and walk down the new tree. I think this is in alignment with the expected experience.

Reply:
This might be a premature optimization. When nodes have stable `id` values, VS Code does a good job of requesting child nodes only when necessary.